File: scratch/test_api.py
import urllib.request
import json
import xml.etree.ElementTree as ET

isbn = "9784042224314"

# OpenBD
try:
    req = urllib.request.urlopen(f"https://api.openbd.jp/v1/get?isbn={isbn}")
    data = json.loads(req.read())
    if data[0]:
        print("OpenBD PublishingDate:", data[0].get("summary", {}).get("pubdate"))
    else:
        print("OpenBD: Not found")
except Exception as e:
    print("OpenBD error:", e)

# NDL
try:
    req = urllib.request.urlopen(f"https://ndlsearch.ndl.go.jp/api/opensearch?isbn={isbn}")
    tree = ET.fromstring(req.read())
    # Find all dc:date
    dates = []
    for elem in tree.iter():
        if 'date' in elem.tag:
            dates.append(elem.text)
    print("NDL dates:", dates)
except Exception as e:
    print("NDL error:", e)

File: scratch/test_api.js
const https = require('https');
const isbn = '9784042224314'; // イリヤの空 その4

// OpenBD
https.get(`https://api.openbd.jp/v1/get?isbn=${isbn}`, (res) => {
    let data = '';
    res.on('data', chunk => data += chunk);
    res.on('end', () => console.log('OpenBD:', data.slice(0, 200)));
});

// Google Books
https.get(`https://www.googleapis.com/books/v1/volumes?q=isbn:${isbn}`, (res) => {
    let data = '';
    res.on('data', chunk => data += chunk);
    res.on('end', () => console.log('Google Books:', JSON.parse(data).items?.[0]?.volumeInfo?.publishedDate));
});

// NDL
https.get(`https://ndlsearch.ndl.go.jp/api/opensearch?isbn=${isbn}`, (res) => {
    let data = '';
    res.on('data', chunk => data += chunk);
    res.on('end', () => {
        const match = data.match(/<dc:date>(.*?)<\/dc:date>/);
        console.log('NDL:', match ? match[1] : 'Not found');
    });
});
